src/main/java/com/mwt/explorers/BootstrapExplorer.java

```diff
@@ -43,22 +43,24 @@ public DecisionTuple chooseAction(long saltedSeed, T context) {
     PRG random = new PRG(saltedSeed);
 
     // Select bag
-    int chosenBag = random.uniformInt(0, policies.size() - 1);
+    final int numPolicies = policies.size();
```

Comment:
Speed improvement to cache the size because it's used multiple times (only improves constant factor in asymptotic runtime).